In [1]:
!pip install ultralytics

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.2/41.2 kB 647.4 kB/s eta 0:00:0000:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 779.6/779.6 kB 6.1 MB/s eta 0:00:00a 0:00:01


In [2]:
pip install -U ipywidgets

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 139.4/139.4 kB 1.3 MB/s eta 0:00:00a 0:00:01m
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 214.4/214.4 kB 3.6 MB/s eta 0:00:00a 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 19.2 MB/s eta 0:00:0000:0100:01
  Attempting uninstall: widgetsnbextension
    Found existing installation: widgetsnbextension 3.6.6
    Uninstalling widgetsnbextension-3.6.6:
      Successfully uninstalled widgetsnbextension-3.6.6
  Attempting uninstall: jupyterlab-widgets
    Found existing installation: jupyterlab-widgets 3.0.9
    Uninstalling jupyterlab-widgets-3.0.9:
      Successfully uninstalled jupyterlab-widgets-3.0.9
  Attempting uninstall: ipywidgets
    Found existing installation: ipywidgets 7.7.1
    Uninstalling ipywidgets-7.7.1:
      Successfully uninstalled ipywidgets-7.7.1
Note: you may need to restart the kernel to use updated packages.


In [3]:
from ultralytics import YOLO

In [ ]:
import os
os.environ['WANDB_MODE'] = 'disabled'

# Train model every fold with yolov8n

training log lost

**fold0**


In [ ]:
model0 = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

results0 = model0.train(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold0.yaml', epochs=100, imgsz=640, batch=8)

**fold1**

In [ ]:
model1 = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

results1 = model1.train(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold1.yaml', epochs=100, imgsz=640, batch=8)

**fold2**

In [ ]:
model2 = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

results2 = model2.train(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold2.yaml', epochs=100, imgsz=640, batch=8)

**fold3**

In [ ]:
model3 = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

results3 = model3.train(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold3.yaml', epochs=100, imgsz=640, batch=8)

**fold4**

In [ ]:
model4 = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

results4 = model4.train(data='/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold4.yaml', epochs=100, imgsz=640, batch=8)

# validation model

**fold0**

In [6]:

# Load a YOLOv8 model
model0 = YOLO('/kaggle/input/test010/best0.pt')
model0.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold0.yaml")

Ultralytics YOLOv8.2.24 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla P100-PCIE-16GB, 16276MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


100%|██████████| 755k/755k [00:00<00:00, 25.5MB/s]
val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/val_fold_0... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:02<00:00, 292.31it/s]

val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:16<00:00,  2.98it/s]


                   all        798       4231      0.686      0.695      0.732      0.554
          hand-raising        798       1744       0.78      0.733      0.805      0.582
               reading        798       1818      0.639      0.697      0.704      0.532
               writing        798        669      0.638      0.655      0.686      0.548
Speed: 0.3ms preprocess, 4.0ms inference, 0.0ms loss, 5.6ms postprocess per image
Results saved to runs/detect/val


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7c1ab0114460>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold1**

In [7]:

# Load a YOLOv8 model
model1 = YOLO('/kaggle/input/test010/best1.pt')
model1.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold1.yaml")

Ultralytics YOLOv8.2.24 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla P100-PCIE-16GB, 16276MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/val_fold_1... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:04<00:00, 192.45it/s]


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:12<00:00,  3.87it/s]


                   all        798       4133      0.677      0.682      0.733      0.548
          hand-raising        798       1902      0.779      0.718      0.802      0.567
               reading        798       1700      0.662      0.676      0.733      0.565
               writing        798        531      0.588      0.652      0.665      0.511
Speed: 0.3ms preprocess, 1.6ms inference, 0.0ms loss, 2.6ms postprocess per image
Results saved to runs/detect/val2


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7c19a681d8d0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold2**

In [8]:
# Load a YOLOv8 model
model2 = YOLO('/kaggle/input/test010/best2.pt')
model2.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold2.yaml")

Ultralytics YOLOv8.2.24 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla P100-PCIE-16GB, 16276MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/val_fold_2... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:04<00:00, 195.28it/s]


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:12<00:00,  4.03it/s]


                   all        798       4109      0.663      0.707      0.729      0.551
          hand-raising        798       1737      0.767      0.763      0.809       0.59
               reading        798       1813      0.639      0.725      0.736      0.559
               writing        798        559      0.582      0.631      0.641      0.504
Speed: 0.3ms preprocess, 1.6ms inference, 0.0ms loss, 2.4ms postprocess per image
Results saved to runs/detect/val3


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7c199419de40>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold3**

In [9]:
# Load a YOLOv8 model
model3 = YOLO('/kaggle/input/test010/best3.pt')
model3.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold3.yaml")

Ultralytics YOLOv8.2.24 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla P100-PCIE-16GB, 16276MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/val_fold_3... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:03<00:00, 221.05it/s]


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:12<00:00,  4.04it/s]


                   all        798       4109      0.669      0.686      0.718      0.538
          hand-raising        798       1775      0.759      0.742      0.791       0.57
               reading        798       1720      0.641      0.694      0.705      0.537
               writing        798        614      0.608      0.624      0.657      0.506
Speed: 0.4ms preprocess, 1.6ms inference, 0.0ms loss, 1.2ms postprocess per image
Results saved to runs/detect/val4


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7c198cfb52d0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold4**

In [10]:
# Load a YOLOv8 model
model4 = YOLO('/kaggle/input/test010/best4.pt')
model4.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_fold4.yaml")

Ultralytics YOLOv8.2.24 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla P100-PCIE-16GB, 16276MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/val_fold_4... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:03<00:00, 216.00it/s]


val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:12<00:00,  4.01it/s]


                   all        798       4204      0.685      0.683      0.721      0.544
          hand-raising        798       1875      0.768      0.707      0.805      0.587
               reading        798       1682      0.611      0.714      0.693      0.534
               writing        798        647      0.675      0.627      0.665      0.512
Speed: 0.3ms preprocess, 1.6ms inference, 0.0ms loss, 2.9ms postprocess per image
Results saved to runs/detect/val5


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7c19a681d8a0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

# Test model to see result

**fold0**

In [4]:
# Load a YOLOv8 model
model00 = YOLO('/kaggle/input/test010/best0.pt')
model00.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.2.28 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


100%|██████████| 755k/755k [00:00<00:00, 14.5MB/s]
val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:02<00:00, 355.06it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:18<00:00,  3.47it/s]


                   all       1026       5426      0.655      0.657      0.701      0.525
          hand-raising        753       2310      0.774      0.743      0.813      0.589
               reading        378       2433      0.641      0.693      0.724      0.552
               writing        193        683      0.551      0.535      0.566      0.435
Speed: 0.3ms preprocess, 4.2ms inference, 0.0ms loss, 1.9ms postprocess per image
Results saved to runs/detect/val


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7adb4358cb50>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold1**

In [5]:
# Load a YOLOv8 model
model01 = YOLO('/kaggle/input/test010/best1.pt')
model01.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.2.28 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 795.87it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:17<00:00,  3.78it/s]


                   all       1026       5426      0.631      0.684      0.702      0.525
          hand-raising        753       2310      0.754      0.743      0.804      0.586
               reading        378       2433      0.632      0.732      0.729      0.553
               writing        193        683      0.508      0.578      0.575      0.438
Speed: 0.3ms preprocess, 2.9ms inference, 0.0ms loss, 3.8ms postprocess per image
Results saved to runs/detect/val2


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7adb10a36f80>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold2**

In [6]:
# Load a YOLOv8 model
model02 = YOLO('/kaggle/input/test010/best2.pt')
model02.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.2.28 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 988.58it/s] 

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:17<00:00,  3.74it/s]


                   all       1026       5426      0.653      0.681      0.708      0.532
          hand-raising        753       2310      0.773       0.74      0.811      0.592
               reading        378       2433      0.638      0.741      0.739      0.567
               writing        193        683      0.549      0.563      0.574      0.436
Speed: 0.2ms preprocess, 2.4ms inference, 0.0ms loss, 2.8ms postprocess per image
Results saved to runs/detect/val3


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7adb0e73f5b0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold3**

In [7]:
# Load a YOLOv8 model
model03 = YOLO('/kaggle/input/test010/best3.pt')
model03.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.2.28 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:01<00:00, 968.35it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:16<00:00,  3.83it/s]


                   all       1026       5426      0.683      0.655       0.71      0.532
          hand-raising        753       2310      0.792       0.73      0.814      0.592
               reading        378       2433      0.663      0.685       0.73      0.556
               writing        193        683      0.595      0.551      0.588      0.448
Speed: 0.3ms preprocess, 4.1ms inference, 0.0ms loss, 0.8ms postprocess per image
Results saved to runs/detect/val4


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7adb0d2cdf30>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

**fold4**

In [8]:
# Load a YOLOv8 model
model04 = YOLO('/kaggle/input/test010/best4.pt')
model04.val(data="/kaggle/input/detecting-student-classroom-behavior2/kaggle_config_test.yaml")

Ultralytics YOLOv8.2.28 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels/test... 1026 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1026/1026 [00:00<00:00, 1183.61it/s]

val: WARNING ⚠️ /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/images/test/3005055.jpg: 1 duplicate labels removed
val: WARNING ⚠️ Cache directory /kaggle/input/detecting-student-classroom-behavior2/data_test2/data_test2/labels is not writeable, cache not saved.



                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 65/65 [00:16<00:00,  3.86it/s]


                   all       1026       5426      0.638      0.687      0.701      0.523
          hand-raising        753       2310      0.753      0.751      0.805      0.584
               reading        378       2433      0.595      0.768      0.724      0.555
               writing        193        683      0.566      0.542      0.573      0.429
Speed: 0.3ms preprocess, 2.5ms inference, 0.0ms loss, 1.1ms postprocess per image
Results saved to runs/detect/val5


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1, 2])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x7adaec62edd0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04

# summary

* fold0
> * 		Class     Images  Instances      Box(P          R      mAP50  mAP50-95)
> *                    all       1026       5426      0.655      0.657      0.701      0.525
> *           hand-raising        753       2310      0.774      0.743      0.813      0.589
> *                reading        378       2433      0.641      0.693      0.724      0.552
> *                writing        193        683      0.551      0.535      0.566      0.435
* **
* **
* fold1
> * 		Class     Images  Instances      Box(P          R      mAP50  mAP50-95)
> *                    all       1026       5426      0.631      0.684      0.702      0.525
> *           hand-raising        753       2310      0.754      0.743      0.804      0.586
> *                reading        378       2433      0.632      0.732      0.729      0.553
> *                writing        193        683      0.508      0.578      0.575      0.438
* **
* **
* fold2
> * 		Class     Images  Instances      Box(P          R      mAP50  mAP50-95)
> *                    all       1026       5426      0.653      0.681      0.708      0.532
> *           hand-raising        753       2310      0.773       0.74      0.811      0.592
> *                reading        378       2433      0.638      0.741      0.739      0.567
> *                writing        193        683      0.549      0.563      0.574      0.436
* **
* **
* fold3
> * 		Class     Images  Instances      Box(P          R      mAP50  mAP50-95)
> *                    all       1026       5426      0.683      0.655       0.71      0.532
> *           hand-raising        753       2310      0.792       0.73      0.814      0.592
> *                reading        378       2433      0.663      0.685       0.73      0.556
> *                writing        193        683      0.595      0.551      0.588      0.448
* **
* **
* fold4
> * 		Class     Images  Instances      Box(P          R      mAP50  mAP50-95)
> *                    all       1026       5426      0.638      0.687      0.701      0.523
> *           hand-raising        753       2310      0.753      0.751      0.805      0.584
> *                reading        378       2433      0.595      0.768      0.724      0.555
> *                writing        193        683      0.566      0.542      0.573      0.429
* **

Analysis

* Fold 0: Has a balanced but lower precision and recall compared to some other folds. mAP50 and mAP50-95 are moderate.
* Fold 1: Shows the lowest precision and mAP50-95 tied with Fold 0. However, it has the highest recall among all folds.
* Fold 2: Has balanced performance with moderate precision, recall, and mAP scores.
* Fold 3: Demonstrates the highest precision and mAP50 with moderate recall.
* Fold 4: Has the second-lowest precision, moderate recall, and the lowest mAP50-95.